lib/wovnrb.rb

```diff
@@ -66,22 +68,26 @@ def call(env)
 
     def switch_lang(headers, body)
       translated_body = []
+
+      # Must use `.each` for to support multiple-chunks in Sinatra
       string_body = ''
       body.each { |chunk| string_body += chunk }
       html_body = Helpers::NokogumboHelper::parse_html(string_body)
 
       if !wovn_ignored?(html_body) && !amp?(html_body)
-        if html_body.html?
-          # TODO: set lang property to HTML
-          # TODO: insert hreflangs
-          # TODO: insert fallback snippet
-        end
+        html_converter = HtmlConverter.new(string_body, @store, headers)
```

Comment:
Why not send `html_body`? If so we can avoid parsing a second time in `HtmlConverter#build_api_compatible_html`

Reply:
Indeed, both string body and dom are needed in HtmlConverter, and I were reluctant to pass them both. Do you think it's good to move `html?` (from wovnrb.rb) to HtmlConverter, so we could avoid parsing 2 times?

Reply:
I prefer it like it is currently. `wovnrb` know when to do nothing, add snippet and use the api; `HtmlConverter` knows only how to edit the HTML; and `ApiTranslator` knows only how to send and read data to and from our API. I think concerns are well-separated and it is easy to understand the big picture by just looking at `Wovnrb#switch_lang`.

Reply:
You might want to look at my other comment here: https://github.com/WOVNio/wovnrb/pull/130#discussion_r229154901

Reply:
ok, I will check. Thanks